src/main/java/ch/njol/skript/expressions/ExprAnvilRepairCost.java

```diff
@@ -0,0 +1,126 @@
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.AnvilInventory;
+import org.bukkit.inventory.Inventory;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Anvil Repair Cost")
+@Description({
+		"Returns the experience cost (in levels) to complete the current repair or the maximum experience cost (in levels) to be allowed by the current repair.",
+		"The default value of max cost set by vanilla Minecraft is 40."
+})
+@Examples({
+		"on inventory click:",
+		"\tif {AnvilRepairSaleActive} = true:",
+		"\t\twait a tick # recommended, to avoid client bugs",
+		"\t\tset anvil repair cost to anvil repair cost * 50%",
+		"\t\tsend \"Anvil repair sale is ON!\" to player",
+
+		"on inventory click:",
+		"\tplayer have permission \"anvil.repair.max.bypass\"",
+		"\tset max repair cost of event-inventory to 99999"
+})
+@Since("INSERT VERSION")
+public class ExprAnvilRepairCost extends SimplePropertyExpression<Inventory, Integer> {
+
+	static {
+		Skript.registerExpression(ExprAnvilRepairCost.class, Integer.class, ExpressionType.PROPERTY,
+			"[the] [anvil] [item] [:max[imum]] repair cost [of %inventories%]",
+			"%inventories%'[s] [item] [:max[imum]] repair cost");
+	}
+
+	boolean isMax = false;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		isMax = parseResult.hasTag("max");
+		setExpr((Expression<? extends Inventory>) exprs[0]);
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	@Nullable
+	public Integer convert(Inventory inventory) {
+		if (!(inventory instanceof AnvilInventory))
+			return null;
+
+		AnvilInventory anvilInventory = (AnvilInventory) inventory;
+		return isMax ? anvilInventory.getMaximumRepairCost() : anvilInventory.getRepairCost();
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case ADD:
+			case REMOVE:
+			case SET:
+				return CollectionUtils.array(Number.class);
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (delta == null)
+			return;
+
```

Comment:
```suggestion
	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
```

Reply:
Why?

Reply:
Might be because it should be impossible for delta to ever be null, but I do agree this seems like a weird change to make

Reply:
We always check for delta null in change method for safety, in addition to that the acceptChange method only accept change modes that require delta not to be null.

Reply:
Skript performs this check before calling change() so it's actually okay to remove it :)